tests/helpers/test_script.py

```diff
@@ -1155,208 +757,127 @@ def record_event(event):
             {"event": event},
         ]
     )
+    script_obj = script.Script(hass, sequence, script_mode=script_mode)
 
-    for run_mode in _ALL_RUN_MODES:
-        events = []
-        hass.states.async_set("test.entity", "hello")
-
-        if run_mode is None:
-            script_obj = script.Script(hass, schema)
-        else:
-            script_obj = script.Script(hass, schema, run_mode=run_mode)
-
-        assert not script_obj.can_cancel
+    assert script_obj.can_cancel == (script_mode != "legacy")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5164426)